plugins/backend/catalog-backend-module-okta/src/providers/OktaUserEntityProvider.ts

```diff
@@ -80,9 +87,22 @@ export class OktaUserEntityProvider extends OktaEntityProvider {
     const allUsers = await client.listUsers({ search: this.userFilter });
 
     await allUsers.each(user => {
+      const profileAnnotations: Record<string, string> = {};
```

Comment:
We can refactor this into a function and reuse the function in three places.

e.g.

Can we refactor this into a function to reduce the triplication.

I think the function could take a user/group. e.g.

```typescript
export getCustomAnnotations(member: User | Group, allowList: string[]) {
  const profileAnnotations: Record<string, string> = {};
  for (const [key, value] of new Map(Object.entries(member.profile))) {
    const stringKey = key.toString();
    if (this.customAttributesToAnnotationAllowlist.includes(stringKey)) {
      profileAnnotations[stringKey] = value.toString();
    }
  }
  return profileAnnotations;
}
```

Then you can just do the following:

```typescript
      const annotations = {
        ...defaultAnnotations,
        ...(getCustomAnnotations(user)),
      };
```